tool-base/src/test/kotlin/io/spine/tools/java/fs/SourceFileSpec.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2022, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package io.spine.tools.java.fs
+
+import com.google.common.testing.NullPointerTester
+import com.google.common.truth.Truth.assertThat
+import com.google.protobuf.DescriptorProtos.DescriptorProto
+import com.google.protobuf.DescriptorProtos.EnumDescriptorProto
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto
+import com.google.protobuf.DescriptorProtos.ServiceDescriptorProto
+import com.google.protobuf.Descriptors.Descriptor
+import com.google.protobuf.Descriptors.EnumDescriptor
+import com.google.protobuf.Descriptors.FileDescriptor
+import com.google.protobuf.Descriptors.ServiceDescriptor
+import com.google.protobuf.Empty
+import io.grpc.protobuf.ProtoServiceDescriptorSupplier
+import io.spine.option.EntityOption
+import io.spine.test.code.NoOuterClassnameSourceFileTest.NoOuterClassnameMessage
+import io.spine.test.code.SourceFile.NestedMessage
+import io.spine.test.code.StandaloneMessage
+import io.spine.testing.setDefault
+import io.spine.tools.type.ProjectServiceGrpc
+import io.spine.type.MessageType
+import java.nio.file.Paths
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import spine.test.code.InheritAllSourceFileTest.InheritAllMessage
+import spine.test.code.InheritPackage
+
+@DisplayName("Java `SourceFile` should point to a generated message class which")
```

Comment:
Can we use this type for non-generated source files?

If so, I wouldn't restrict the description to just that.

Reply:
Right. Fixed.